### PR TITLE
Using updated slackutil in knative/pkg

### PR DESCRIPTION
**Pull request by Fredy-Z**
**What this PR does, why we need it**:
- Use updated slackutil in https://github.com/knative/pkg/pull/645.
- Use `helpers.Run` and `helpers.CombineErrors` in https://github.com/knative/pkg/pull/645.

### CODE REVIEW

**Comment by chaodaiG** [comment]
It surprised me that the build-test passed when there are changes under vendor directly, seems like a bug

**Comment by srinivashegde86** [comment]
Use the helpers for `https://github.com/knative/test-infra/blob/master/tools/flaky-test-reporter/error.go and https://github.com/knative/test-infra/blob/master/tools/flaky-test-reporter/run.go` as well.

**Comment by Fredy-Z** [comment]
> It surprised me that the build-test passed when there are changes under vendor directly, seems like a bug

Yeah we run `./hack/update-deps.sh` on it, but we don't `diff` the result like what's done in `/hack/verify-codegen.sh`.

**Comment by Fredy-Z** [comment]
> Use the helpers for `https://github.com/knative/test-infra/blob/master/tools/flaky-test-reporter/error.go and https://github.com/knative/test-infra/blob/master/tools/flaky-test-reporter/run.go` as well.

Thanks for the reminder! Will do it.

**Comment by Fredy-Z** [comment]
> It surprised me that the build-test passed when there are changes under vendor directly, seems like a bug

Yeah it's not preventing such change, since the `default_build_test_runner` does not run `./hack/update-deps.sh`.

File: tools/flaky-test-reporter/slack_notification.go [hunk 1]
```diff
@@ -84,7 +84,8 @@ func sendSlackNotifications(repoDataAll []RepoData, c slackutil.WriteOperations,
 		for i := range channels {
 			wg.Add(1)
 			channel := channels[i]
-			go func(wg *sync.WaitGroup) {
+			go func() {
```
Comment (chaodaiG) [inline tools/flaky-test-reporter/slack_notification.go]: are you sure this works without passing wg?

**Comment by chaodaiG** [comment]
> > It surprised me that the build-test passed when there are changes under vendor directly, seems like a bug
> 
> Yeah it's not preventing such change, since the `default_build_test_runner` does not run `./hack/update-deps.sh`.

it's because we don't have `verify-codegen.sh` in test-infra

File: tools/flaky-test-reporter/slack_notification.go (hunk 1 above)
Comment (Fredy-Z) [reply tools/flaky-test-reporter/slack_notification.go]: Yes, one example here - https://golang.org/src/sync/example_test.go

**Comment by Fredy-Z** [comment]
> > > It surprised me that the build-test passed when there are changes under vendor directly, seems like a bug
> > 
> > 
> > Yeah it's not preventing such change, since the `default_build_test_runner` does not run `./hack/update-deps.sh`.
> 
> it's because we don't have `verify-codegen.sh` in test-infra

`verify-codegen.sh` pairs with `update-codegen.sh`, we don't have any code to generate, so..

**Comment by chaodaiG** [comment]
> > > > It surprised me that the build-test passed when there are changes under vendor directly, seems like a bug
> > > 
> > > 
> > > Yeah it's not preventing such change, since the `default_build_test_runner` does not run `./hack/update-deps.sh`.
> > 
> > 
> > it's because we don't have `verify-codegen.sh` in test-infra
> 
> `verify-codegen.sh` pairs with `update-codegen.sh`, we don't have any code to generate, so..

not entirely true, at least we have this to generate: https://github.com/knative/serving/blob/24bc31214c6140a9d03570974b414082d004fa9b/hack/update-codegen.sh#L83

**Comment by Fredy-Z** [comment]
> > > > > It surprised me that the build-test passed when there are changes under vendor directly, seems like a bug
> > > > 
> > > > 
> > > > Yeah it's not preventing such change, since the `default_build_test_runner` does not run `./hack/update-deps.sh`.
> > > 
> > > 
> > > it's because we don't have `verify-codegen.sh` in test-infra
> > 
> > 
> > `verify-codegen.sh` pairs with `update-codegen.sh`, we don't have any code to generate, so..
> 
> not entirely true, at least we have this to generate: https://github.com/knative/serving/blob/24bc31214c6140a9d03570974b414082d004fa9b/hack/update-codegen.sh#L83

Ah yeah, that's true, didn't notice that..

File: tools/flaky-test-reporter/json_reporter.go [hunk 2]
```diff
@@ -55,7 +56,7 @@ func writeFlakyTestsToJSON(repoDataAll []RepoData, dryrun bool) error {
 			for test := range testSet {
 				testList = append(testList, test)
 			}
-			if err := run(
+			if err := helpers.Run(
```
Comment (srinivashegde86) [inline tools/flaky-test-reporter/json_reporter.go]: Since you are using the helpers, can we delete these two files from here?

Comment (Fredy-Z) [reply tools/flaky-test-reporter/json_reporter.go]: I did delete them. Github diff "intelligently" considered them as renames :-(

Comment (srinivashegde86) [reply tools/flaky-test-reporter/json_reporter.go]: oh. okies

**Pull request merged**